authority/linkedca.go

```diff
@@ -0,0 +1,490 @@
+package authority
+
+import (
+	"context"
+	"crypto"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/db"
+	"go.step.sm/crypto/jose"
+	"go.step.sm/crypto/keyutil"
+	"go.step.sm/crypto/tlsutil"
+	"go.step.sm/crypto/x509util"
+	"go.step.sm/linkedca"
+	"golang.org/x/crypto/ssh"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const uuidPattern = "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
+
+type linkedCaClient struct {
+	renewer     *tlsutil.Renewer
+	client      linkedca.MajordomoClient
+	authorityID string
+}
+
+type linkedCAClaims struct {
+	jose.Claims
+	SANs []string `json:"sans"`
+	SHA  string   `json:"sha"`
+}
+
+func newLinkedCAClient(token string) (*linkedCaClient, error) {
+	tok, err := jose.ParseSigned(token)
```

Comment:
Do we verify this token at some point?

Reply:
We cannot verify this token at this point because we don't have access to the public key. This token is used to bootstrap the mTLS certificate between the linked ca and majordomo.